components/server/ee/src/user/eligibility-service.ts

```diff
@@ -270,7 +269,26 @@ export class EligibilityService {
             Plans.TEAM_PROFESSIONAL_USD,
         ].map(p => p.chargebeeId);
 
-        return subscriptions.filter(s => eligblePlans.includes(s.planId!)).length > 0;
+        const relevantSubscriptions = subscriptions.filter(s => eligblePlans.includes(s.planId!));
+        if (relevantSubscriptions.length === 0) {
+            // user has no subscription that grants "more ressources"
+            return false;
+        }
+
+        // some TeamSubscriptions are marked with 'noMoreResources' to convey that those are _not_ receiving more resources
+        const noMoreResources = await Promise.all(relevantSubscriptions.map(async (s): Promise<boolean> => {
+            if (!s.teamSubscriptionSlotId) {
+                return false;
+            }
+            const ts = await this.teamSubscriptionDb.findTeamSubscriptionBySlotId(s.teamSubscriptionSlotId);
+            return !!ts?.noMoreResources;
```

Comment:
Thought: I feel like this could be more "elegant" a second filter right after the `eligiblePlans.includes(s.planId)` above, thus leading to a single `if (relevantSubscriptions.length === 0) return false` (i.e. without needing a `noMoreResources` interim variable or a `noMoreResources.every()` call)

Reply:
This is certainly partly a subjective thing. The problem here is that due to TS lagging proper combinators for async (like `filterAsync`), you'd need to encode both filters as `map` returning `Subscription | undefined`, followed by a `every` (or `filter` + length), which I feel is counterintuitive. So I decided to:
 - keep the filter where it feels natural
 - only use the complex `map` for async stuff